compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1345,7 +1346,7 @@ TR::Instruction *OMR::Power::MemoryReference::expandInstruction(TR::Instruction
             }
```

Comment:
It looks like the code above may have issues when `cg->hiValue(displacement) == 0x8000`.

Reply:
Added compensation code.